src/parser.rs

```diff
@@ -469,6 +469,20 @@ impl Parser {
         }
     }
 
+    /// Parse the kinds of things that can be fed to EXTRACT and DATE_TRUNC
+    pub fn parse_extract_field(&mut self) -> Result<ExtractField, ParserError> {
```

Comment:
Unimportant nit: `parse_date_time_field` (above) and `parse_date_time_field_given_str`(below) are lightly tied together if you wanted to move this either above or below both.